File: fel-adapter/src/de/proskor/fel/impl/EventInstanceImpl.scala
package de.proskor.fel.impl

import de.proskor.automation.Element
import de.proskor.fel.container.EventInstanceContainer
import de.proskor.fel.event.EventInstance
import de.proskor.fel.event.EventType

class EventInstanceImpl(peer: Element) extends EntityImpl(peer) with EventInstance {
  override def getContainer: EventInstanceContainer = {
    new EventInstanceContainerImpl(peer.parent.get)
  }

  override def getType: EventType = {
    val types = for {
      connector <- peer.connectors
      if connector.source == peer && connector.stereotype == "instanceOf"
      target = connector.target
    } yield new EventTypeImpl(target)
    types.headOption.orNull
  }

  override def setType(typ: EventType) {
    val connector = peer.connectors.add("", "Connector"); connector.stereotype = "instanceOf"
    connector.source = peer
    connector.target = typ.asInstanceOf[EventTypeImpl].peer
  }
}

File: ea-fel/src/de/proskor/fel/stub/EventInstanceImpl.java
package de.proskor.fel.stub;

import de.proskor.fel.container.EventInstanceContainer;
import de.proskor.fel.event.EventInstance;
import de.proskor.fel.event.EventType;

public class EventInstanceImpl extends EntityImpl implements EventInstance {
	private EventInstanceContainer container;
	private EventType typ;

	@Override
	public EventInstanceContainer getContainer() {
		return this.container;
	}

	public void setContainer(EventInstanceContainer container) {
		this.container = container;
	}

	@Override
	public EventType getType() {
		return this.typ;
	}

	@Override
	public void setType(EventType typ) {
		this.typ = typ;
	}
}
